nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -1189,4 +1180,17 @@ public List<ConfigVerificationResult> verify(final ProcessContext context, final
 
         return results;
     }
+
+    /**
+     * Return whether to include the entity in the listing, based on the minimum and maximum object age (if configured).
+     */
+    private boolean includeObjectInListing(final ProcessContext context, final S3VersionSummary versionSummary) {
+        final Long minAgeMilliseconds = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+        final Long maxAgeMilliseconds = context.getProperty(MAX_AGE) != null ? context.getProperty(MAX_AGE).asTimePeriod(TimeUnit.MILLISECONDS) : null;
```

Comment:
I feel a bit overkill to evaluate the properties for each listed item separately. As Min/Max Age do not change while the processor is running, they can be stored in instance fields of the processor object that can be set in `onScheduled()`.